internal/connectors/plugins/public/column/client/timeline.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+// Timeline allows the client to navigate the backlog and decide whether to fetch
+// historical or recently added data
+type Timeline struct {
+	LatestID      string `json:"latest_id"`
+	BacklogCursor string `json:"backlog_cursor"`
+}
+
+func (t Timeline) IsCaughtUp() bool {
+	return t.LatestID != ""
+}
+
+func (c *client) scanForOldest(
+	ctx context.Context,
+	timeline Timeline,
+	endpoint string,
+	pageSize int,
+) (*Transaction, Timeline, bool, error) {
+	req, err := c.newRequest(ctx, http.MethodGet, endpoint, http.NoBody)
+	if err != nil {
+		return nil, timeline, false, fmt.Errorf("failed to create timeline transactions request: %w", err)
+	}
+
+	q := req.URL.Query()
+	q.Add("limit", strconv.Itoa(pageSize))
+	if timeline.BacklogCursor != "" {
+		q.Add("starting_after", timeline.BacklogCursor)
+	}
+	req.URL.RawQuery = q.Encode()
+
+	var res TransactionResponseWrapper[[]*Transaction]
+	var errRes columnError
+	_, err = c.httpClient.Do(ctx, req, &res, &errRes)
+	if err != nil {
+		return nil, timeline, false, fmt.Errorf("failed to get transactions for timeline: %w %w", err, errRes.Error())
+	}
+
+	if len(res.Transfers) == 0 {
+		return nil, timeline, res.HasMore, nil
+	}
+
+	oldest := res.Transfers[len(res.Transfers)-1]
+	if !res.HasMore {
+		timeline.LatestID = oldest.ID
```

Comment:
is it the oldest or the latest? 🤯

Reply:
I kept the same nomenclature as the Stripe Timeline.

Because Column's API is terrible:
* when we fetch with `starting_after` we are scrolling newest to oldest
* when we fetch with `ending_before` we are scrolling oldest to newest (in general) but Column still returns the results newest to oldest, so we flip the results and save the newest ID from this batch.

So by `LatestID` I mean both "newest" (since we are scrolling old to new) and something along the lines of "last seen ID".

Reply:
LatestID makes sense, but `oldest` still is confusing to me? Can we rename that transfer to lastSeen?

Reply:
In this case it is always the oldest! The `scanTimeline` function only deals with going back in time so the last item in the list is always the oldest within the scope of this function.

I set the `LatestID` as the oldest item because I want the caller to start its very first query using the ID of the oldest record in the PSP's database (basically we've rewound to the beginning of time, so the latest payment is also the oldest payment.

I agree it is a bit confusing I'll see what I can do to make it more obvious...